pdns/dnswriter.cc

```diff
@@ -325,7 +325,7 @@ void DNSPacketWriter::xfrName(const DNSName& name, bool compress, bool)
 
   uint16_t li=0;
   uint16_t matchlen=0;
-  if(compress && (li=lookupName(name, &matchlen))) {
+  if(compress && (li=lookupName(name, &matchlen)) && li < 16384) {
```

Comment:
I wonder if that check should live in `DNSPacketWriter::lookupName()` instead so we can try to find an existing name with less matching labels?

Reply:
ah yes

Reply:
pushed